src/Tests/dotnet-test.Tests/GivenDotnetTestContainsEnvironmentVariables.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using System.IO;
+using System;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Xml;
+using System.Xml.Linq;
+using System.Reflection;
+using dotnet.Tests;
+using System.Runtime.InteropServices;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit.Abstractions;
+using Microsoft.NET.TestFramework.Utilities;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Cli.Test.Tests
+{
+    public class GivenDotnetTestContainsEnvironmentVariables : SdkTest
+    {
+        private const string TestAppName = "VSTestEnvironmentVariables";
+        private const string EnvironmentOption = "--environment";
+        private const string EnvironmentVariable1 = "__DOTNET_TEST_ENVIRONMENT_VARIABLE_EMPTY";
+        private const string EnvironmentVariable2 = "__DOTNET_TEST_ENVIRONMENT_VARIABLE_1=VALUE1";
+        private const string EnvironmentVariable3 = "__DOTNET_TEST_ENVIRONMENT_VARIABLE_2=VALUE WITH SPACE";
```

Comment:
@KathleenDollard the syntax is a bit odd. _dotnet test --environment  key1=value1 --environment  key2=value2_

But we don't have something like this before.

Reply:
This is to allow setting env variable just for the target process in unified way. On linux you can already do it this way, but not on windows: 

```
key1=value1 key2=value2 dotnet test
```

So we just take that syntax and use -e or --environment to do the same thing cross platform. Here is my reasoning about the UI https://github.com/microsoft/vstest/issues/669#issuecomment-578813697, and here is the reasoning behind the feature https://github.com/microsoft/vstest/issues/669#issuecomment-578773872. 

We already use a similar syntax for inline settings where you can do `dotnet test -- option1=value1 parent.option2=value2` this get's translated to nodes in the runsettings.xml.